src/utils/atomWithStorage.ts

```diff
@@ -148,15 +156,22 @@ export function atomWithStorage<Value>(
 
   const anAtom = atom(
     (get) => get(baseAtom),
-    (get, set, update: SetStateAction<Value> | typeof RESET) => {
+    (get, set, update: StorageSetStateAction<Value> | typeof RESET) => {
       if (update === RESET) {
         set(baseAtom, initialValue)
         return storage.removeItem(key)
       }
+
       const newValue =
         typeof update === 'function'
-          ? (update as (prev: Value) => Value)(get(baseAtom))
+          ? (update as SetStateActionFunc<Value>)(get(baseAtom))
```

Comment:
With the above change, this would become:
```suggestion
          ? (update as (prev: Value) => Value | typeof RESET)(get(baseAtom))
```